iris_ugrid/tests/test_regrid.py

```diff
@@ -0,0 +1,177 @@
+# Copyright Iris-ugrid contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+Test ESMF regridding.
+"""
+
+import numpy as np
+from numpy import ma
+from iris_ugrid.regrid import MeshInfo, GridInfo, Regridder
+import scipy.sparse
+
+
+def make_small_mesh_args():
+    ugrid_node_coords = np.array([[0, 0], [0, 1], [1, 0], [1, 1], [1, 2]])
+    ugrid_face_node_connectivity = ma.array(
+        [[0, 2, 3, -1], [3, 0, 1, 4]],
+        mask=np.array([[0, 0, 0, 1], [0, 0, 0, 0]]),
+    )
+    return ugrid_node_coords, ugrid_face_node_connectivity
+
+
+def make_small_grid_args():
+    small_x = 2
+    small_y = 3
+    small_grid_lon = np.array(range(small_x)) / (small_x + 1)
+    small_grid_lat = np.array(range(small_y)) * 2 / (small_y + 1)
+
+    small_grid_lon_bounds = np.array(range(small_x + 1)) / (small_x + 1)
+    small_grid_lat_bounds = np.array(range(small_y + 1)) * 2 / (small_y + 1)
+    return (
+        small_grid_lon,
+        small_grid_lat,
+        small_grid_lon_bounds,
+        small_grid_lat_bounds,
+    )
+
+
+def expected_weights():
+    weight_list = np.array(
+        [
+            0.3333836384685291,
+            0.6666163615314712,
+            1.0,
+            0.3335106008404508,
+            0.9167189586203999,
+            0.0832810413795998,
+            0.08339316630404843,
+            0.9166068336959516,
+            0.9168310094376751,
+        ]
+    )
+    rows = np.array([0, 0, 1, 2, 3, 3, 4, 4, 5])
+    columns = np.array([0, 1, 1, 1, 0, 1, 0, 1, 1])
+
+    shape = (6, 2)
+
+    weights = scipy.sparse.csr_matrix(
+        (weight_list, (rows, columns)), shape=shape
+    )
+    return weights
+
+
+def test_make_mesh():
+    coords, nodes = make_small_mesh_args()
+    mesh = MeshInfo(coords, nodes, 0)
+    ESMF_mesh_0 = mesh.make_esmf_field()
```

Comment:
It's not good style to start a local variable name with capitals, because that is usually used to indicate a class : I guess black may pick you up on this ?

Likewise for `ESMF_mesh_0` below, and `ESMF_grid` in the following test.